### PR TITLE
RED-184 - CLI fixes for BN

### PR DESCRIPTION
**Pull request by Glitch18**
The status command is broken and needs fixes for how values are handled.

### CODE REVIEW

File: src/node-commands.ts [hunk 1]
```diff
@@ -437,7 +437,7 @@ export function registerNodeCommands(program: Command) {
           yaml.dump({
             stake: eoaData?.operatorAccountInfo
               ? ethers.utils.formatEther(
-                  String(parseInt(eoaData.operatorAccountInfo.stake, 16))
+                  String(parseInt(eoaData.operatorAccountInfo.stake.value, 16))
```
Comment (mssabr01) [inline src/node-commands.ts]: What happens if stake and value do not exist on this object?

File: src/utils/fetch-network-data.ts [hunk 2]
```diff
@@ -160,8 +160,14 @@ export async function fetchInitialParameters(
...
-      nodeRewardAmount: new BN(parsedValue.nodeRewardAmount, 16),
-      nodeRewardInterval: new BN(parsedValue.nodeRewardInterval, 16),
+      nodeRewardAmount: new BN(
+        stripHexPrefix(parsedValue.nodeRewardAmount),
+        16
+      ),
+      nodeRewardInterval: new BN(
+        stripHexPrefix(parsedValue.nodeRewardInterval),
+        16
+      ),
```
Comment (mssabr01) [inline src/utils/fetch-network-data.ts]: Same here. nodeRewardAmount might not exist

File: src/utils/fetch-network-data.ts [hunk 3]
```diff
@@ -175,8 +181,11 @@ export async function fetchInitialParameters(
   if (!response) {
     throw new Error("Fetched initial parameters, but account data isn't found");
   }
-  const nodeRewardAmount = new BN(response.nodeRewardAmountUsd, 16);
-  const nodeRewardInterval = new BN(response.nodeRewardInterval, 16);
+  const nodeRewardAmount = new BN(
+    stripHexPrefix(response.nodeRewardAmountUsd),
+    16
+  );
+  const nodeRewardInterval = new BN(response.nodeRewardInterval);
```
Comment (mssabr01) [inline src/utils/fetch-network-data.ts:190]: Catch if nodeRewardAmountUsd and nodeRewardInterval does not exist

File: src/utils/fetch-network-data.ts [hunk 4]
```diff
@@ -354,8 +363,11 @@ export async function getAccountInfoParams(
       return params;
     }
 
-    params.lockedStake = new BN(nodeData.stakeLock, 16).toString();
-    previousRewards = new BN(nodeData.reward, 16);
+    params.lockedStake = new BN(
```
Comment (mssabr01) [inline src/utils/fetch-network-data.ts]: Please check that stakeLock is present

**Review by mssabr01 — DISMISSED** [review]

File: src/node-commands.ts (hunk 1 above)
Comment (ahmxdiqbal) [reply src/node-commands.ts]: its wrapped in a try catch, so it would've just console logged the error. now it should just log a empty string

**Comment by Glitch18** [comment]
@mssabr01 Changes by Ahmed look good. Let me know if there are any further changes required before this can be merged

**Review by achal-singh — APPROVED** [review]

**Review by mssabr01 — COMMENTED** [review]
lgtm

**Pull request merged**